src/Swarm/Language/Syntax.hs

```diff
@@ -86,27 +90,59 @@ import Swarm.Language.Types
 -- | The type of directions. Used /e.g./ to indicate which way a robot
 --   will turn.
 data Direction = Lft | Rgt | Back | Fwd | North | South | East | West | Down
-  deriving (Eq, Ord, Show, Read, Generic, Data, Hashable, ToJSON, FromJSON)
+  deriving (Eq, Ord, Show, Read, Generic, Data, Hashable, ToJSON, FromJSON, Enum, Bounded)
 
 instance ToJSONKey Direction where
   toJSONKey = genericToJSONKey defaultJSONKeyOptions
 
 instance FromJSONKey Direction where
   fromJSONKey = genericFromJSONKey defaultJSONKeyOptions
 
+data DirInfo = DirInfo
+  { dirSyntax :: Text
+  , -- absolute direction if it exists
+    dirAbs :: Maybe (V2 Int64)
+  , -- the turning for the direction
+    dirApplyTurn :: V2 Int64 -> V2 Int64
+  }
+
+allDirs :: [Direction]
+allDirs = [minBound .. maxBound]
+
+-- | TODO Information about all directions
+dirInfo :: Direction -> DirInfo
+dirInfo d = case d of
+  Lft -> DirInfo "left" Nothing (\(V2 x y) -> V2 (- y) x)
```

Comment:
Maybe we could prefix direction with `D`, so then you could do the same `lowShow` trick as with `Const` and avoid having to write `South -> DirInfo "south"`.

Reply:
Ohk so it needs a "D" prefix like `DLeft`, `DRight` because Prelude exports `Left` and `Right`